packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts

```diff
@@ -191,3 +232,107 @@ describeCompat("TestSignals", "FullCompat", (getTestObjectProvider) => {
 		);
 	});
 });
+
+describeCompat("Targeted Signals", "NoCompat", (getTestObjectProvider) => {
+	const numberOfClients = 3;
+	assert(numberOfClients >= 2, "Need at least 2 clients for targeted signals");
+	let clients: SignalClient[];
+	let provider: ITestObjectProvider;
+
+	beforeEach("setup containers", async () => {
+		provider = getTestObjectProvider();
+		clients = [];
+		for (let i = 0; i < numberOfClients; i++) {
+			const container = await (i === 0
+				? provider.makeTestContainer(testContainerConfig)
+				: provider.loadTestContainer(testContainerConfig));
+			const dataObject = await getContainerEntryPointBackCompat<ITestFluidObject>(container);
+
+			if (container.connectionState !== ConnectionState.Connected) {
+				await new Promise((resolve) => container.once("connected", resolve));
+			}
+
+			clients.push({
+				dataStoreRuntime: dataObject.runtime,
+				containerRuntime: dataObject.context.containerRuntime,
+				signalReceivedCount: 0,
+				clientId: container.clientId,
+			});
+		}
+	});
+	async function sendAndVerifyRemoteSignals(runtime: RuntimeLayer) {
+		clients.forEach((client) => {
+			client[runtime].on("signal", (message: IInboundSignalMessage, local: boolean) => {
+				assert.equal(local, false, "Signal should be remote");
+				assert.equal(message.type, "TestSignal", "Signal type should be TestSignal");
+				assert.equal(message.content, true, "Signal content should be true");
+				assert.equal(
+					message.targetClientId,
+					client.clientId,
+					"Signal should be targeted to this client",
+				);
+				client.signalReceivedCount += 1;
+			});
+		});
+
+		for (let i = 0; i < clients.length; i++) {
+			const targetClient = clients[(i + 1) % clients.length];
+			clients[i][runtime].submitSignal("TestSignal", true, targetClient.clientId);
+			await waitForTargetedSignal(
+				targetClient[runtime],
+				clients.filter((c) => c !== targetClient).map((c) => c[runtime]),
+			);
+		}
+
+		clients.forEach((client, index) => {
+			assert.equal(
+				client.signalReceivedCount,
+				1,
+				`client ${index + 1} did not receive signal`,
+			);
+		});
+	}
+	async function sendAndVerifyLocalSignals(runtime: RuntimeLayer) {
+		clients.forEach((client) => {
+			client[runtime].on("signal", (message: IInboundSignalMessage, local: boolean) => {
+				assert.equal(local, true, "Signal should be local");
```

Comment:
This is the test that has me really wondering what is being tested.
local should be true only when a client receives its own message (at least that is my understanding). So, unless a client is targeting itself, it should not be receiving a message with local = true.

Reply:
Yes that's my understanding too, this function sendAndVerifyLocalSignals is strictly for testing local targeted signals (clients targeting themselves).

Reply:
I see. So, it is the "remote" vs "local" that terms that I didn't understand. "other" and "self" would be clearer (to me). "remote" isn't so bad. "local" is a problem because it is unclear if there is any messaging happening outside of the local client. Note the name of the test file too - what does that use of "local" mean?
Also, lets change the `true` in the middle of:
```
client[runtime].submitSignal("TestSignal", true, client.clientId);
```
It should be something unambiguous.

Reply:
Ok, I swapped out the usage of "local" for "self" and also improved the naming quality of test signal content and type to remove ambiguity.